VSTSDropProxy.cs

```diff
@@ -159,59 +122,74 @@ await Policy
+            var downloads = uniquefiles.Select(async group => 
             {
-                if (!file.Value.Paths.Any())
-                {
-                    // should this be DropException?
-                    throw new ArgumentException($"empty : {file.Key}");
-                }
-                
-                var firstPath = file.Value.Paths.First();
-                var localFilename = firstPath.Substring(_relativeroot.Length);
-                var localPath = Path.Combine(localDestiantion,localFilename).Replace("\\","/");
-                await Download(file.Value.Url, localPath);
+                var f = group.First();
+                var relativepath = f.Path.Substring(_relativeroot.Length);
+                var localPath = Path.Combine(localDestiantion,relativepath).Replace("\\","/");
+                await Download(f.Blob.Url, localPath);
                  
                 // parallelize this too? worth it?
-                foreach (var other in file.Value.Paths.Skip(1))
+                foreach (var other in group.Skip(1))
                 {
-                    var otherFileName = other.Substring(_relativeroot.Length);
-                    var otherpath = Path.Combine(localDestiantion,otherFileName).Replace("\\","/");
+                    var otherrelativepath = other.Path.Substring(_relativeroot.Length);
+                    var otherpath = Path.Combine(localDestiantion,otherrelativepath).Replace("\\","/");
                     File.Copy(localPath, otherpath);
                 }
                 if (++downloaded % 100 == 0)
                 {
                     Console.WriteLine($"Downloaded {downloaded} files");
-                }
-                
+                }                
             });
-
             await Task.WhenAll(downloads);
         } 
-    }
-
-    struct UniqueFile
-    {
-        public string Url;
-        public List<string> Paths;
+      
+        private string HashFiles(IEnumerable<VstsFile> files)
+        {
+            var hasher = new System.Security.Cryptography.SHA256Managed();
+            hasher.Initialize();
+            foreach( var f in files.OrderBy(f => f.Path))
+            {
+                var buffer = Encoding.UTF8.GetBytes(f.Blob.Id);
+                hasher.TransformBlock(buffer, 0, buffer.Length, null, 0);
+                buffer = Encoding.UTF8.GetBytes(f.Path);
+                hasher.TransformBlock(buffer, 0, buffer.Length, null, 0);
+            }
+            hasher.TransformFinalBlock(new byte[0], 0, 0);
+            return System.BitConverter.ToString(hasher.Hash).Replace("-","");
+        }
     }
 
     // Helper classes for parsing VSTS drop exe output lowercase to match json output
     public sealed class VstsFile
     {
         public string Path { get; set; }
         public VstsBlob Blob { get; set; }
+
+        public override int GetHashCode() { return StringComparer.OrdinalIgnoreCase.GetHashCode(Path); }
```

Comment:
You should also override Equals if you're overriding GetHashCode here.

Reply:
Also, do you *want* case-insensitive filenames? Is that a problem on the Unix side at all? Something we want to care about or nah?

Reply:
Original code worked in both linux and windows. Some drops did have duplicate files of different case. Not a problem for aether but might be for others. Could remove this and the replacces above if this was full linux but meh.